nemo/collections/nlp/data/intent_slot_classification/intent_slot_classification_dataset.py

```diff
@@ -0,0 +1,272 @@
+
+    Args:
+        input_file (str): file to sequence + label.
+            the first line is header (sentence [tab] label)
+            each line should be [sentence][tab][label]
+        slot_file (str): file to slot labels, each line corresponding to
+            slot labels for a sentence in input_file. No header.
+        max_seq_length (int): max sequence length minus 2 for [CLS] and [SEP]
+        tokenizer (Tokenizer): such as NemoBertTokenizer
+        num_samples (int): number of samples you want to use for the dataset.
+            If -1, use all dataset. Useful for testing.
+        pad_label (int): pad value use for slot labels.
+            by default, it's the neutral label.
+
+    """
+
+    def __init__(
+        self,
+        input_file,
+        slot_file,
+        max_seq_length,
+        tokenizer,
+        num_samples=-1,
+        pad_label=128,
+        ignore_extra_tokens=False,
+        ignore_start_end=False,
+        do_lower_case=False,
+    ):
+        if num_samples == 0:
+            raise ValueError("num_samples has to be positive", num_samples)
+
+        with open(slot_file, 'r') as f:
+            slot_lines = f.readlines()
+
+        with open(input_file, 'r') as f:
+            input_lines = f.readlines()[1:]
+
+        assert len(slot_lines) == len(input_lines)
+
+        dataset = list(zip(slot_lines, input_lines))
+
+        if num_samples > 0:
+            dataset = dataset[:num_samples]
+
+        raw_slots, queries, raw_intents = [], [], []
+        for slot_line, input_line in dataset:
+            raw_slots.append([int(slot) for slot in slot_line.strip().split()])
+            parts = input_line.strip().split()
+            raw_intents.append(int(parts[-1]))
+            query = ' '.join(parts[:-1])
+            if do_lower_case:
+                query = query.lower()
+            queries.append(query)
+
+        features = get_features(
+            queries,
+            max_seq_length,
+            tokenizer,
+            pad_label=pad_label,
+            raw_slots=raw_slots,
+            ignore_extra_tokens=ignore_extra_tokens,
+            ignore_start_end=ignore_start_end,
+        )
+        self.all_input_ids = features[0]
+        self.all_segment_ids = features[1]
+        self.all_input_mask = features[2]
+        self.all_loss_mask = features[3]
+        self.all_subtokens_mask = features[4]
+        self.all_slots = features[5]
+        self.all_intents = raw_intents
+
+    def __len__(self):
+        return len(self.all_input_ids)
+
+    def __getitem__(self, idx):
+        return (
+            np.array(self.all_input_ids[idx]),
+            np.array(self.all_segment_ids[idx]),
+            np.array(self.all_input_mask[idx], dtype=np.long),
+            np.array(self.all_loss_mask[idx]),
+            np.array(self.all_subtokens_mask[idx]),
+            self.all_intents[idx],
+            np.array(self.all_slots[idx]),
+        )
+
+
+class IntentSlotInferenceDataset(Dataset):
```

Comment:
Dataset should be typed in the new NeMo. Please check the latest text classifciation dataset.

Reply:
Removed this class for now